### ISSUE TITLE
Sync client is stuck in an endless uploading loop when syncing large files to a Federated Share

### ISSUE THREAD

**Issue by mmattel** [owncloud/client#4796]
**Please see core issue https://github.com/owncloud/core/issues/24454 because this one is related to ! 
The system setup ect is described there**

### Expected behaviour
After putting a 165MB video file into a folder that gets synced (folder is on a federated share), the sync should start and complete sucessfully.

### Actual behaviour
Sync starts and upload completes, but during the waiting time while the owncloud server uploads to the federated share, the client runes into a timeout and restarts the upload. This confuses the system completely, because this is a endless loop ! The old upload to the federated share has not finished and a new upload from the sync client with the same file starts conflicting with the cached file.
To stop this loop, you have to take out the large file from the folder to be synced - BUT when you logon in the browser (after a while) the large file shows up (because it has been uploaded to the federated owncloud system successfully) and now gets back synced to the client (I also have seen that the large file was visible in the browser only but not in the synced folder...)

### Server configuration
Operating system: Ubuntu 14.04

Web server: nginx

Database: mysql

PHP version: 5.5.9

ownCloud version:
ownCloud 9.0.2 RC1 (daily) Build:2016-04-28T03:03:12+00:00 795f321a1985b682bff584f252dd6f2fd95dad65

Storage backend:

### Client configuration
Client version:
2.2.0beta2 (build 6044)
(same with the stable version)

Operating system: W7x64

OS language: German

Installation path of client: standard

### Logs

1. Client: cant share as this contains sensitive data. But it is easy to reproduce...
(There should be a anonymizing function for the log output like you have in core for config.php)

2. Web server error log:
no corresponding errors

3. ownCloud log (data/owncloud.log):
````
Exception: {"Exception":"OCP\\Lock\\LockedException","Message":"\"FederatedShareFolder\/TestVideo.mp4\" is locked","Code":0,"Trace":"#0 \/var\/www\/owncloud\/lib\/private\/files\/view.php(1956): OC\\Files\\View->lockPath('\/files\/FederatedS...', 1, false)\
n#1 \/var\/www\/owncloud\/lib\/private\/files\/view.php(803): OC\\Files\\View->lockFile('\/files\/FederatedS...', 1)\
n#2 \/var\/www\/owncloud\/apps\/files_versions\/lib\/storage.php(193): OC\\Files\\View->copy('files\/\/FederatedS...', 'files_versions\/...')\
n#3 \/var\/www\/owncloud\/apps\/files_versions\/lib\/hooks.php(60): OCA\\Files_Versions\\Storage::store('\/FederatedSon Eng...')\
n#4 [internal function]: OCA\\Files_Versions\\Hooks::write_hook(Array)\
n#5 \/var\/www\/owncloud\/lib\/private\/hook.php(105): call_user_func(Array, Array)\
n#6 \/var\/www\/owncloud\/lib\/private\/filechunking.php(215): OC_Hook::emit('OC_Filesystem', 'write', Array)\
n#7 \/var\/www\/owncloud\/apps\/dav\/lib\/connector\/sabre\/file.php(455): OC_FileChunking->file_assemble(Object(OCA\\Files_Trashbin\\Storage), 'SubFolder1\/SubSubFolder1...', '\/TestUser\/files...')\
n#8 \/var\/www\/owncloud\/apps\/dav\/lib\/connector\/sabre\/file.php(103): OCA\\DAV\\Connector\\Sabre\\File->createFileChunked(Resource id #48)\
n#9 \/var\/www\/owncloud\/apps\/dav\/lib\/connector\/sabre\/directory.php(134): OCA\\DAV\\Connector\\Sabre\\File->put(Resource id #48)\
n#10 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(1036): OCA\\DAV\\Connector\\Sabre\\Directory->createFile(TestVideo -...', Resource id #48)\
n#11 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php(523): Sabre\\DAV\\Server->createFile('FederatedS...', Resource id #48, NULL)\
n#12 [internal function]: Sabre\\DAV\\CorePlugin->httpPut(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\
n#13 \/var\/www\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\
n#14 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(459): Sabre\\Event\\EventEmitter->emit('method:PUT', Array)\
n#15 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(248): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\
n#16 \/var\/www\/owncloud\/apps\/dav\/appinfo\/v1\/webdav.php(55): Sabre\\DAV\\Server->exec()\
n#17 \/var\/www\/owncloud\/remote.php(138): require_once('\/var\/w...')\
n#18 {main}","File":"\/var\/www\/owncloud\/lib\/private\/files\/view.php","Line":1858}
````

**Comment by guruz** [comment]
FYI @icewind1991 @PVince81 @DeepDiver1975

**Comment by PVince81** [comment]
@guruz I commented in the server ticket. We need to improve upload perf.

On the client side, might need to find a way to better handle timeouts.
Maybe the new chunking endpoint might help.

**Comment by guruz** [comment]
Thanks!
Indeed, streaming uploads would be good here..

**Comment by dragotin** [comment]
Wait, why do you say "Endless upload loop"? This problem should be blacklisted, which means, that the client tries to upload again (hoping that the problem is temporar) but with increasing time spans in between.

Doesn't that happen for you?

**Comment by mmattel** [comment]
I did not measure the time inbetween.  But having the 165MB file as soucre to be synced and looking that the upload started again for the 3rd time, I did no furthier investigation because it started to "kill" my server and logged long error messages in my oc-log. The only chance to interrupt the upload was to move the file out of the sync folder and do manual cleanups.

Referencing https://github.com/owncloud/core/issues/12097 (Asynchronous PUT).
BTW: this thread has started Nov 2014...

It is not the filesize as absolute value, but the combination of file size, sync upload time, backend upload time ect. And as more these values drift apart, as worse the situation gets...
The only thing I can say is, that the current behaviour for this situation is an absoulte no go :bomb:
